txpool/pool.go

```diff
@@ -2231,16 +2231,27 @@ func (mt *metaTx) better(than *metaTx, pendingBaseFee uint256.Int) bool {
 		if effectiveTip.Cmp(&thanEffectiveTip) != 0 {
 			return effectiveTip.Cmp(&thanEffectiveTip) > 0
 		}
+		// Only compare nonces and balances for same sender
+		if mt.Tx.SenderID == than.Tx.SenderID {
+			if mt.nonceDistance != than.nonceDistance {
+				return mt.nonceDistance > than.nonceDistance
+			}
+			if mt.cumulativeBalanceDistance != than.cumulativeBalanceDistance {
+				return mt.cumulativeBalanceDistance > than.cumulativeBalanceDistance
+			}
+		}
```

Comment:
are you sure we don't need `else { ` branch here? Let's think about it a bit

Reply:
Also, I copied the thing wrongly. The comparison is incorrect for `better()` function. Fixing it.

Reply:
> Oh interesting point. My initial thought was that it only makes sense to compare nonces if we're dealing with same sender. But now if I see, the field is nonce distance and not the nonce actually. It makes sense to put a tx with less nonce distance first then one with more nonce distance even though they're from different sender. Same applies to cumulativeBalanceDistance. Am I correct?

yes. It's inter-sender-calculated-field :-) 
Yes, for general-purpose txpool better avoid "preferring one sender over another" - and just try to maximize miner fee profit.
But as i remember we "preferring local transactions (submitted by user) over others".

Reply:
Got it, fixed.